src/components/ViewResponses/Export.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import {Button, Row} from "antd";
+
+interface Response {
+    answerType: number
+    inputValue: string
+    otherMessage: string
+    question: string
+    tag: string
+}
+
+interface UserInfo {
+    [pubkey: string]: {
+        display_name: string
+    }
+}
+
+export const Export: React.FC<{responses: {plaintext: string, pubkey: string}[], userInfo: any}> = (props) => {
+    const onDownloadClick = async (type: 'csv' | 'excel') => {
+        const XLSX = await import('xlsx')
+
+        const responses = props.responses.map(({plaintext}) => JSON.parse(plaintext)) as Response[][]
+        const parsedResponse = responses
+            .map((response, index) => {
+                const resp = {
+                    User: props.userInfo[props.responses[index].pubkey]?.display_name ?? 'Anonymous Response'
+                }
+                response
+                    .reduce<Record<string, string>>((newResponse, answer) => {
+                        newResponse[answer.question] = answer.inputValue
+                        return newResponse
+                    }, resp)
+                return resp
+            })
+        const workSheet = XLSX.utils.json_to_sheet(parsedResponse)
+        const workBook = XLSX.utils.book_new()
+        XLSX.utils.book_append_sheet(workBook, workSheet, 'Responses')
+        if(type === 'excel') {
+            XLSX.writeFile(workBook, 'Responses.xlsx')
+        } else {
+            XLSX.writeFile(workBook, 'Responses.csv')
```

Comment:
should this contain the npub of the form? or atleast name of the form? Responses, could be a little meaningless

Reply:
yes agreed.

not this though. The worksheet should have the name of the form but the name of this sheet can be the name of the form.

that way we can also give the feature to export data of all forms, so 1 excel can have multiple worksheets corresponding to different forms